tracer/test/Datadog.Trace.IntegrationTests/SpanTagTests.cs

```diff
@@ -144,5 +117,143 @@ public void SpanSampler_ShouldTagMultiple_OnSpanFinish()
+            }
+        }
+
+        [Fact]
+        public void SpanSampler_ShouldNotTag_WhenSpansAreKept()
+        {
+            var ruleRate = "1";
+            var maxPerSecond = "1000";
+            var samplingMechanism = "8";
+
+            using (var rootScope = _tracer.StartActive("root"))
+            {
+                using (var childScope = _tracer.StartActive("child1"))
+                {
+                }
+
+                using (var childScope = _tracer.StartActive("child2"))
+                {
+                }
+            }
+
+            var traces = _testApi.Wait();
+            traces.Should().HaveCount(1); // 1 trace...
+            traces[0].Should().HaveCount(3); // ...with 3 spans
+
+            var rootSpan = traces[0].SingleOrDefault(s => s.ParentId is null or 0)!;
+            rootSpan.Should().NotBeNull();
+
+            // assert that root span has the span sampling tags
+            rootSpan.Tags.Should().NotContain(Tags.SingleSpanSampling.RuleRate, ruleRate);
+            rootSpan.Tags.Should().NotContain(Tags.SingleSpanSampling.MaxPerSecond, maxPerSecond);
+            rootSpan.Tags.Should().NotContain(Tags.SingleSpanSampling.SamplingMechanism, samplingMechanism);
+
+            // assert child spans have span sampling tags
+            var childSpans = traces[0].Where(s => s.ParentId is not null and not 0);
+
+            foreach (var span in childSpans)
+            {
+                span.Tags.Should().NotContain(Tags.SingleSpanSampling.RuleRate, ruleRate);
+                span.Tags.Should().NotContain(Tags.SingleSpanSampling.MaxPerSecond, maxPerSecond);
+                span.Tags.Should().NotContain(Tags.SingleSpanSampling.SamplingMechanism, samplingMechanism);
+            }
+        }
+
+        [Fact]
+        public void SpanSampler_ShouldNotTag_WhenTraceContext_IsNull()
+        {
+            var expectedRuleRate = "1";
+            var expectedMaxPerSecond = "1000";
+            var expectedSamplingMechanism = "8";
+
+            var inputSpan = new Span(new SpanContext(5, 6, null, serviceName: "test"), DateTimeOffset.Now) { OperationName = "test" };
+            inputSpan.Context.TraceContext.Should().BeNull();
+            var spans = new Span[1];
+            spans[0] = inputSpan;
+            _writer.WriteTrace(new ArraySegment<Span>(spans));
+
+            var trace = _testApi.Wait();
+            trace.Should().HaveCount(1);
+            trace[0].Should().HaveCount(1);
+
+            var span = trace[0].Single();
+            span.Tags.Should().NotContain(Tags.SingleSpanSampling.RuleRate, expectedRuleRate);
+            span.Tags.Should().NotContain(Tags.SingleSpanSampling.MaxPerSecond, expectedMaxPerSecond);
+            span.Tags.Should().NotContain(Tags.SingleSpanSampling.SamplingMechanism, expectedSamplingMechanism);
+        }
+
+        [Fact]
+        public void SpanSampler_ShouldNotTag_WhenSamplingPriority_IsNull()
```

Comment:
For this test, I'd like to test a scenario where we start a span with the `SpanContext` object. I think that use-case would be much more common where the sampling priority could be null

Reply:
Does the change in bdb52c835f8a7d5a533229944aaae880bfa577ab do this?